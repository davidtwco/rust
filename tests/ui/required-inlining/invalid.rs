//@ compile-flags: --crate-type=lib
#![allow(internal_features)]
#![feature(rustc_attrs)]
#![feature(required_inlining)]

// Test that invalid required/must inlining attributes error as expected.

#[inline(required = "bar")]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(required)]`
pub fn reqd1() {
}

#[inline(required(bar, baz))]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(required)]`
pub fn reqd2() {
}

#[inline(required(2))]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(required)]`
pub fn reqd3() {
}

#[inline(required = 2)]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(required)]`
pub fn reqd4() {
}

#[inline(must = "bar")]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(must)]`
pub fn must1() {
}

#[inline(must(bar, baz))]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(must)]`
pub fn must2() {
}

#[inline(must(2))]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(must)]`
pub fn must3() {
}

#[inline(must = 2)]
//~^ ERROR invalid argument
//~^^ HELP expected one string argument to `#[inline(must)]`
pub fn must4() {
}

#[inline(banana)]
//~^ ERROR invalid argument
//~^^ HELP valid inline arguments are `required`, `must`, `always` and `never`
pub fn bare() {
}
