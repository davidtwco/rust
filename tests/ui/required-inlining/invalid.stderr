error[E0535]: invalid argument
  --> $DIR/invalid.rs:8:1
   |
LL | #[inline(required = "bar")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(required)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:14:1
   |
LL | #[inline(required(bar, baz))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(required)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:20:1
   |
LL | #[inline(required(2))]
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(required)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:26:1
   |
LL | #[inline(required = 2)]
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(required)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:32:1
   |
LL | #[inline(must = "bar")]
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(must)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:38:1
   |
LL | #[inline(must(bar, baz))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(must)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:44:1
   |
LL | #[inline(must(2))]
   | ^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(must)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:50:1
   |
LL | #[inline(must = 2)]
   | ^^^^^^^^^^^^^^^^^^^
   |
   = help: expected one string argument to `#[inline(must)]`

error[E0535]: invalid argument
  --> $DIR/invalid.rs:56:10
   |
LL | #[inline(banana)]
   |          ^^^^^^
   |
   = help: valid inline arguments are `required`, `must`, `always` and `never`

error: aborting due to 9 previous errors

For more information about this error, try `rustc --explain E0535`.
